src/test/java/de/zalando/aruha/nakadi/partitioning/OrderingKeyFieldsPartitioningStrategyTest.java

```diff
@@ -0,0 +1,187 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import org.json.JSONObject;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import java.util.TreeSet;
+import java.util.function.Consumer;
+
+import static de.zalando.aruha.nakadi.utils.TestUtils.resourceAsString;
+import static java.lang.Integer.parseInt;
+import static java.lang.Math.abs;
+import static java.lang.Math.pow;
+import static java.util.Arrays.asList;
+import static java.util.Arrays.stream;
+import static java.util.stream.Collectors.toCollection;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Stream.generate;
+import static org.apache.commons.lang3.RandomStringUtils.randomAlphabetic;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.lessThan;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertThat;
+
+public class OrderingKeyFieldsPartitioningStrategyTest {
+
+    public static final Random RANDOM = new Random();
+    public static final String DELIMITER = "#";
+    public static final int NUMBER_OF_PARTITIONS = 8;
+
+    private final OrderingKeyFieldsPartitioningStrategy strategy = new OrderingKeyFieldsPartitioningStrategy();
+    private final EventType simpleEventType;
+    private final ArrayList<List<JSONObject>> partitions = createEmptyPartitions(NUMBER_OF_PARTITIONS);
+
+    public OrderingKeyFieldsPartitioningStrategyTest() {
+        simpleEventType = new EventType();
+        simpleEventType.setOrderingKeyFields(asList("sku", "name"));
+    }
+
+    @Test
+    public void calculatesSamePartitionForSameOrderingKeyFields() throws Exception {
+        fillPartitionsWithEvents(simpleEventType, partitions, 1000);
+
+        checkThatEventsWithSameKeysAreInSamePartition(partitions);
+    }
+
+    @Test
+    public void partitionsAreEquallyDistributed() {
+        fillPartitionsWithEvents(simpleEventType, partitions, 10000);
+
+        final double[] eventDistribution = partitions.stream().map(p -> p.size()).mapToDouble(value -> value * 1.0).toArray();
+        final double variance = calculateVarianceOfUniformDistribution(eventDistribution);
+
+        assertThat(variance, lessThan(1.5));
```

Comment:
It looks like an approachable problem from a statistical point of view. But tackling this problem would add not too much value to the project and since I'm no expert I'll just take it.

If we knew the sample variance distribution, we could choose a safe interval where the build would fail with controlled estimated frequency.